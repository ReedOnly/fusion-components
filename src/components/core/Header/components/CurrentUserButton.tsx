import React from 'react';
import { useCurrentUser, useFusionContext } from '@equinor/fusion';
import { useIcon, IconProps, MenuSection, MenuItemType, Menu, usePopoverRef } from 'index';

const CurrentUserIcon = (props: IconProps) => {
    const iconFactory = useIcon(
        <path
            fillRule="evenodd"
            clipRule="evenodd"
            d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM12 5C13.66 5 15 6.34 15 8C15 9.66 13.66 11 12 11C10.34 11 9 9.66 9 8C9 6.34 10.34 5 12 5ZM6 15.98C7.29 17.92 9.5 19.2 12 19.2C14.5 19.2 16.71 17.92 18 15.98C17.97 13.99 13.99 12.9 12 12.9C10 12.9 6.03 13.99 6 15.98Z"
            fill="currentColor"
        />
    );

    return iconFactory(props);
};

const CurrentUserDropdown: React.FC = () => {
    const { auth } = useFusionContext();
    const currentUser = useCurrentUser();

    if (!currentUser) {
        return null;
    }

    const sections: MenuSection[] = [
        {
            key: currentUser.id,
            title: `${currentUser.givenName} ${currentUser.familyName}`,
            items: [
                {
                    key: 'logout',
                    title: 'Sign out',
                },
            ],
            // TODO: Allow core and apps to add custom buttons like "Preview features" etc.
            // Should be provided through the fusion context, not props
        },
    ];

    const onClick = async (item: MenuItemType) => {
        switch (item.key) {
            case 'logout':
                await auth.container.logoutAsync();
                break;
        }
    };

    return <Menu sections={sections} onClick={onClick} elevation={0} />;
};

const CurrentUserButton: React.FC = () => {
    const currentUser = useCurrentUser();
    const popoverRef = usePopoverRef<HTMLButtonElement>(<CurrentUserDropdown />, {
        placement: 'below',
        justify: 'end',
        centered: true,
        fillWithContent: true,
    });

    if (!currentUser) {
        return null;
    }

    return (
        <button ref={popoverRef}>
            <CurrentUserIcon />
        </button>
    );
};

export default CurrentUserButton;
