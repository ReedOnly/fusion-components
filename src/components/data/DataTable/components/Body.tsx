import React from 'react';
import { DataTableBodyProps } from '../types';
import Row from './Row';
import { getString, getBoolean } from '../utils';

function DataTableBody<T>({
    data,
    columns,
    collapsedColumns,
    rowIdentifier,
    isExpandable,
    expandedComponent,
    onExpand,
    expandedItems,
}: DataTableBodyProps<T>) {
    return (
        <>
            {data.map((item, index) => (
                <Row
                    key={getString(item, rowIdentifier)}
                    item={item}
                    columns={columns}
                    collapsedColumns={collapsedColumns}
                    index={index}
                    isExpandable={isExpandable ? getBoolean(item, isExpandable) : false}
                    isExpanded={expandedItems.findIndex(i => i === item) > -1}
                    expandedComponent={expandedComponent}
                    onExpand={() => onExpand(item, index)}
                />
            ))}
        </>
    );
}

export default DataTableBody;
