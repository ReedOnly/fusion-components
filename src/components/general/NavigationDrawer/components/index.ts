export { default as MenuChild } from "./Child";