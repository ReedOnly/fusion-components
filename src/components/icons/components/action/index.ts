export { default as SearchIcon } from './SearchIcon';
export { default as DoneIcon } from './DoneIcon';
export { default as MinimizeIcon } from './MinimizeIcon';
