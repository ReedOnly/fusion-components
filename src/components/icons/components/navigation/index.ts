export { default as PaginationArrow } from "./PaginationArrow";
export { default as CloseIcon } from "./CloseIcon";