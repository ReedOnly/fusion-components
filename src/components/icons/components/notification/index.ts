export { default as SyncDisabledIcon } from './SyncDisabledIcon';
