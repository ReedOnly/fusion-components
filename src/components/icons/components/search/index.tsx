import * as React from 'react';
import { IconProps, useIcon } from 'index';

const SearchIcon = (props: IconProps) => {
    const iconFactory = useIcon(
        <path
            fillRule="evenodd"
            clipRule="evenodd"
            d="M12.1865 11.4716H12.9765L17.2165 15.7316C17.6265 16.1416 17.6265 16.8116 17.2165 17.2216C16.8065 17.6316 16.1365 17.6316 15.7265 17.2216L11.4765 12.9716V12.1816L11.2065 11.9016C9.80646 13.1016 7.89646 13.7216 5.86646 13.3816C3.08646 12.9116 0.866456 10.5916 0.526456 7.79157C0.00645569 3.56157 3.56646 0.00157288 7.79646 0.521573C10.5965 0.861573 12.9165 3.08157 13.3865 5.86157C13.7265 7.89157 13.1065 9.80157 11.9065 11.2016L12.1865 11.4716ZM2.47646 6.97157C2.47646 9.46157 4.48646 11.4716 6.97646 11.4716C9.46646 11.4716 11.4765 9.46157 11.4765 6.97157C11.4765 4.48157 9.46646 2.47157 6.97646 2.47157C4.48646 2.47157 2.47646 4.48157 2.47646 6.97157Z"
            fill="currentColor"
        />
    );

    return iconFactory(props);
};

export default SearchIcon;
