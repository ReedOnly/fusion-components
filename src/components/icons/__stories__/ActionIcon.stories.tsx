import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { actions } from '@storybook/addon-actions';
import { withKnobs, number, color, select } from '@storybook/addon-knobs';
import withFusionStory from "../../../../.storybook/withFusionStory";

import { SearchIcon, SortIcon } from '../components/action';

const eventsFromNames = actions('onClick');

const stories = storiesOf('Icons|Action', module);
stories.addDecorator(withKnobs);
stories.addDecorator(withFusionStory("Action"));

stories.add('Search', () => (
    <SearchIcon
        {...eventsFromNames}
        width={number('width', 24)}
        height={number('height', 24)}
        color={color('color', '#000')}
    />
));

stories.add('Sort', () => (
    <SortIcon
        {...eventsFromNames}
        direction={select('Direction', {
            None: null,
            Ascending: 'asc',
            Descending: 'desc',
        })}
    />
));
