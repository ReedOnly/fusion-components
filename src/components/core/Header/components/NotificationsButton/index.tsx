import * as React from 'react';
import * as styles from './styles.less';
import classNames from 'classnames';
import { useComponentDisplayClassNames, useNotificationCards } from '@equinor/fusion';
import NotificationCardsPresenter from '../../../NotificationCardsPresenter';
import useIcon, { IconProps } from 'hooks/useIcon';
import styling from 'styles/styling';
import IconButton from 'components/general/IconButton';
import { NotificationsIcon } from 'components/icons/components/notification';
import NotificationsSideSheet from 'components/general/NotificationsSideSheet';
import OverlayPortal from 'components/utils/OverlayPortal';

type NotificationNumberBadgeProps = IconProps & {
    numberOfUnread: number;
};

const NotificationNumberBadge = (props: NotificationNumberBadgeProps) => {
    const iconFactory = useIcon(
        <>
            <circle cx="12" cy="12" r="12" fill={styling.colors.danger} />
            <text x="50%" y="50%" textAnchor="middle" stroke="white" strokeWidth="1px" dy=".3em">
                {props.numberOfUnread}
            </text>
        </>
    );

    return iconFactory(props);
};
const NotificationsButton: React.FC = () => {
    const [showSideSheet, setShowSideSheet] = React.useState<boolean>(false);
    const [hasOpenedSideSheet, setHasOpenedSideSheet] = React.useState<boolean>(false);

    const {
        notificationCards,
        isFetchingRead,
        isFetchingUnRead,
        getReadNotificationCardsAsync,
    } = useNotificationCards();

    const numberOfUnread = React.useMemo(
        () => notificationCards.filter((notification) => !notification.seenByUser).length,
        [notificationCards]
    );
    const badgeClassNames = classNames(
        styles.badgeContainer,
        useComponentDisplayClassNames(styles)
    );

    React.useEffect(() => {
        if (showSideSheet) {
            setHasOpenedSideSheet(true);
        }
    }, [showSideSheet]);

    const openSideSheet = React.useCallback(() => setShowSideSheet(true), [setShowSideSheet]);
    const closeSideSheet = React.useCallback(() => setShowSideSheet(false), [setShowSideSheet]);

    React.useEffect(() => {
        hasOpenedSideSheet && getReadNotificationCardsAsync();
    }, [hasOpenedSideSheet]);

    return (
        <>
            <IconButton onClick={openSideSheet}>
                <div className={styles.notificationsButton}>
                    <NotificationsIcon />
                    {numberOfUnread > 0 && (
                        <div className={badgeClassNames}>
                            <NotificationNumberBadge
                                numberOfUnread={numberOfUnread}
                                width={styling.numericalGrid(3)}
                                height={styling.numericalGrid(3)}
                            />
                        </div>
                    )}
                </div>
            </IconButton>
            <NotificationsSideSheet
                open={showSideSheet}
                onClose={closeSideSheet}
                notifications={notificationCards}
                isFetchingReadNotifications={isFetchingRead}
                isFetchingUnReadNotifications={isFetchingUnRead}
            />
            <OverlayPortal show>
                <NotificationCardsPresenter onShowInList={openSideSheet} />
            </OverlayPortal>
        </>
    );
};

export default NotificationsButton;
