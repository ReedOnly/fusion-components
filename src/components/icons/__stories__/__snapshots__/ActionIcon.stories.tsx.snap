// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons|Action CheckCircle 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <svg
      height={24}
      onClick={[Function]}
      style={
        Object {
          "color": "#000",
        }
      }
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        clipRule="evenodd"
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-2-5.83l6.59-6.59L18 9l-8 8-4-4 1.41-1.41L10 14.17z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`Icons|Action CloseCircle 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <svg
      height={24}
      onClick={[Function]}
      style={
        Object {
          "color": "#000",
        }
      }
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        clipRule="evenodd"
        d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-9.41L15.59 7 17 8.41 13.41 12 17 15.59 15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`Icons|Action Delete 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <svg
      height={24}
      onClick={[Function]}
      style={
        Object {
          "color": "#000",
        }
      }
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        clipRule="evenodd"
        d="M15.5 4H18C18.55 4 19 4.45 19 5C19 5.55 18.55 6 18 6H6C5.45 6 5 5.55 5 5C5 4.45 5.45 4 6 4H8.5L9.21 3.29C9.39 3.11 9.65 3 9.91 3H14.09C14.35 3 14.61 3.11 14.79 3.29L15.5 4ZM8 21C6.9 21 6 20.1 6 19V9C6 7.9 6.9 7 8 7H16C17.1 7 18 7.9 18 9V19C18 20.1 17.1 21 16 21H8Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`Icons|Action Done 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <svg
      height={24}
      onClick={[Function]}
      style={
        Object {
          "color": "#000",
        }
      }
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        clipRule="evenodd"
        d="M8.80004 15.905L5.30004 12.405C5.11519 12.2179 4.86309 12.1125 4.60004 12.1125C4.33699 12.1125 4.0849 12.2179 3.90004 12.405C3.51004 12.795 3.51004 13.415 3.90004 13.805L8.09004 17.995C8.48004 18.385 9.11004 18.385 9.50004 17.995L20.1 7.40501C20.49 7.01501 20.49 6.39501 20.1 6.00501C19.9152 5.81786 19.6631 5.71252 19.4 5.71252C19.137 5.71252 18.8849 5.81786 18.7 6.00501L8.80004 15.905Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`Icons|Action Dropdown 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <svg
      height={24}
      onClick={[Function]}
      style={
        Object {
          "color": "",
        }
      }
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M8.70627 11.4137L11.2963 14.0037C11.6863 14.3937 12.3163 14.3937 12.7063 14.0037L15.2963 11.4137C15.9263 10.7837 15.4763 9.70374 14.5863 9.70374H9.40627C8.51627 9.70374 8.07627 10.7837 8.70627 11.4137Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`Icons|Action Help 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <svg
      height={24}
      onClick={[Function]}
      style={
        Object {
          "color": "#000",
        }
      }
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        clipRule="evenodd"
        d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM11 19V17H13V19H11ZM14.17 12.17L15.07 11.25C16.09 10.23 16.44 8.48 15.26 6.85C14.36 5.6 12.91 4.81 11.39 5.05C9.84 5.29 8.59 6.41 8.16 7.88C8 8.44 8.4 9 8.98 9H9.28C9.67 9 9.98 8.72 10.1 8.35C10.43 7.4 11.46 6.77 12.57 7.08C13.27 7.28 13.83 7.89 13.96 8.61C14.09 9.31 13.87 9.97 13.41 10.41L12.17 11.67C11.75 12.09 11.42 12.61 11.22 13.19C11.08 13.6 11 14.04 11 14.5V15H13C13 14.54 13.05 14.18 13.13 13.86C13.31 13.14 13.67 12.68 14.17 12.17Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`Icons|Action History 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <svg
      height={24}
      onClick={[Function]}
      style={
        Object {
          "color": "#000",
        }
      }
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        clipRule="evenodd"
        d="M4.14407 12.0018C4.14407 6.95175 8.31407 2.86175 13.4041 3.00175C18.0941 3.13175 22.0141 7.05175 22.1441 11.7418C22.2841 16.8318 18.1941 21.0018 13.1441 21.0018C11.0541 21.0018 9.14407 20.2918 7.62407 19.0918C7.15407 18.7318 7.12407 18.0218 7.54407 17.6018C7.90407 17.2418 8.46407 17.2118 8.86407 17.5218C10.0441 18.4518 11.5341 19.0018 13.1441 19.0018C17.0441 19.0018 20.1941 15.8118 20.1441 11.9018C20.0941 8.18175 16.9641 5.05175 13.2441 5.00175C9.32407 4.95175 6.14407 8.10175 6.14407 12.0018H7.93407C8.38407 12.0018 8.60407 12.5417 8.29407 12.8518L5.50407 15.6518C5.30407 15.8518 4.99407 15.8518 4.79407 15.6518L2.00407 12.8518C1.68407 12.5417 1.90407 12.0018 2.35407 12.0018H4.14407ZM12.1441 8.75175C12.1441 8.34175 12.4841 8.00175 12.8941 8.00175C13.3041 8.00175 13.6441 8.34175 13.6441 8.74175V12.1418L16.5241 13.8518C16.8741 14.0618 16.9941 14.5218 16.7841 14.8818C16.5741 15.2318 16.1141 15.3518 15.7541 15.1418L12.6341 13.2917C12.3341 13.1118 12.1441 12.7818 12.1441 12.4318V8.75175Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`Icons|Action Info 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <svg
      height={24}
      onClick={[Function]}
      style={
        Object {
          "color": "#000",
        }
      }
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        clipRule="evenodd"
        d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM12 17C11.45 17 11 16.55 11 16V12C11 11.45 11.45 11 12 11C12.55 11 13 11.45 13 12V16C13 16.55 12.55 17 12 17ZM11 9H13V7H11V9Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`Icons|Action Lock 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <svg
      height={24}
      onClick={[Function]}
      style={
        Object {
          "color": "#000",
        }
      }
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        clipRule="evenodd"
        d="M18 8.5H17V6.5C17 3.74 14.76 1.5 12 1.5C9.24 1.5 7 3.74 7 6.5V8.5H6C4.9 8.5 4 9.4 4 10.5V20.5C4 21.6 4.9 22.5 6 22.5H18C19.1 22.5 20 21.6 20 20.5V10.5C20 9.4 19.1 8.5 18 8.5ZM12 17.5C10.9 17.5 10 16.6 10 15.5C10 14.4 10.9 13.5 12 13.5C13.1 13.5 14 14.4 14 15.5C14 16.6 13.1 17.5 12 17.5ZM9 6.5V8.5H15V6.5C15 4.84 13.66 3.5 12 3.5C10.34 3.5 9 4.84 9 6.5Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`Icons|Action Minimize 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <svg
      height={24}
      onClick={[Function]}
      style={
        Object {
          "color": "#000",
        }
      }
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        clipRule="evenodd"
        d="M7 11H17C17.55 11 18 11.45 18 12C18 12.55 17.55 13 17 13H7C6.45 13 6 12.55 6 12C6 11.45 6.45 11 7 11Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`Icons|Action Open In New 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <svg
      height={24}
      onClick={[Function]}
      style={
        Object {
          "color": "#000",
        }
      }
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        clipRule="evenodd"
        d="M5 18C5 18.55 5.45 19 6 19H18C18.55 19 19 18.55 19 18V13C19 12.45 19.45 12 20 12C20.55 12 21 12.45 21 13V19C21 20.1 20.1 21 19 21H5C3.9 21 3 20.1 3 19V5C3 3.9 3.89 3 5 3H11C11.55 3 12 3.45 12 4C12 4.55 11.55 5 11 5H6C5.45 5 5 5.45 5 6V18ZM15 5C14.45 5 14 4.55 14 4C14 3.45 14.45 3 15 3H20C20.55 3 21 3.45 21 4V9C21 9.55 20.55 10 20 10C19.45 10 19 9.55 19 9V6.41L9.87 15.54C9.48 15.93 8.85 15.93 8.46 15.54C8.07 15.15 8.07 14.52 8.46 14.13L17.59 5H15Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`Icons|Action Print 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <svg
      height={24}
      onClick={[Function]}
      style={
        Object {
          "color": "#000",
        }
      }
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        clipRule="evenodd"
        d="M7 3H17C17.55 3 18 3.45 18 4V6C18 6.55 17.55 7 17 7H7C6.45 7 6 6.55 6 6V4C6 3.45 6.45 3 7 3ZM5 8H19C20.66 8 22 9.34 22 11V15C22 16.1 21.1 17 20 17H18V19C18 20.1 17.1 21 16 21H8C6.9 21 6 20.1 6 19V17H4C2.9 17 2 16.1 2 15V11C2 9.34 3.34 8 5 8ZM9 19H15C15.55 19 16 18.55 16 18V14H8V18C8 18.55 8.45 19 9 19ZM19 12C18.45 12 18 11.55 18 11C18 10.45 18.45 10 19 10C19.55 10 20 10.45 20 11C20 11.55 19.55 12 19 12Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`Icons|Action Schedule 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <svg
      height={24}
      onClick={[Function]}
      style={
        Object {
          "color": "#000",
        }
      }
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        clipRule="evenodd"
        d="M11.99 2C6.47 2 2 6.48 2 12C2 17.52 6.47 22 11.99 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 11.99 2ZM12 20C7.58 20 4 16.42 4 12C4 7.58 7.58 4 12 4C16.42 4 20 7.58 20 12C20 16.42 16.42 20 12 20ZM11.72 7H11.78C12.18 7 12.5 7.32 12.5 7.72V12.26L16.37 14.56C16.72 14.76 16.83 15.21 16.62 15.55C16.42 15.89 15.98 15.99 15.64 15.79L11.49 13.3C11.18 13.12 11 12.79 11 12.44V7.72C11 7.32 11.32 7 11.72 7Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`Icons|Action Search 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <svg
      height={24}
      onClick={[Function]}
      style={
        Object {
          "color": "#000",
        }
      }
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        clipRule="evenodd"
        d="M15.1866 14.4716H15.9766L20.2166 18.7316C20.6266 19.1416 20.6266 19.8116 20.2166 20.2216C19.8066 20.6316 19.1366 20.6316 18.7266 20.2216L14.4766 15.9716V15.1816L14.2066 14.9016C12.8066 16.1016 10.8966 16.7216 8.86658 16.3816C6.08658 15.9116 3.86658 13.5916 3.52658 10.7916C3.00658 6.56156 6.56658 3.00156 10.7966 3.52156C13.5966 3.86156 15.9166 6.08156 16.3866 8.86156C16.7266 10.8916 16.1066 12.8016 14.9066 14.2016L15.1866 14.4716ZM5.47658 9.97156C5.47658 12.4616 7.48658 14.4716 9.97658 14.4716C12.4666 14.4716 14.4766 12.4616 14.4766 9.97156C14.4766 7.48156 12.4666 5.47156 9.97658 5.47156C7.48658 5.47156 5.47658 7.48156 5.47658 9.97156Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`Icons|Action Settings 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <svg
      height={24}
      onClick={[Function]}
      style={
        Object {
          "color": "#000",
        }
      }
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        clipRule="evenodd"
        d="M19.5022 12C19.5022 12.34 19.4722 12.66 19.4322 12.98L21.5422 14.63C21.7322 14.78 21.7822 15.05 21.6622 15.27L19.6622 18.73C19.5422 18.95 19.2822 19.04 19.0522 18.95L16.5622 17.95C16.0422 18.34 15.4822 18.68 14.8722 18.93L14.4922 21.58C14.4622 21.82 14.2522 22 14.0022 22H10.0022C9.75216 22 9.54216 21.82 9.51216 21.58L9.13216 18.93C8.52216 18.68 7.96216 18.35 7.44216 17.95L4.95216 18.95C4.73216 19.03 4.46216 18.95 4.34216 18.73L2.34216 15.27C2.22216 15.05 2.27216 14.78 2.46216 14.63L4.57216 12.98C4.53216 12.66 4.50216 12.33 4.50216 12C4.50216 11.67 4.53216 11.34 4.57216 11.02L2.46216 9.37C2.27216 9.22 2.21216 8.95 2.34216 8.73L4.34216 5.27C4.46216 5.05 4.72216 4.96 4.95216 5.05L7.44216 6.05C7.96216 5.66 8.52216 5.32 9.13216 5.07L9.51216 2.42C9.54216 2.18 9.75216 2 10.0022 2H14.0022C14.2522 2 14.4622 2.18 14.4922 2.42L14.8722 5.07C15.4822 5.32 16.0422 5.65 16.5622 6.05L19.0522 5.05C19.2722 4.97 19.5422 5.05 19.6622 5.27L21.6622 8.73C21.7822 8.95 21.7322 9.22 21.5422 9.37L19.4322 11.02C19.4722 11.34 19.5022 11.66 19.5022 12ZM8.50216 12C8.50216 13.93 10.0722 15.5 12.0022 15.5C13.9322 15.5 15.5022 13.93 15.5022 12C15.5022 10.07 13.9322 8.5 12.0022 8.5C10.0722 8.5 8.50216 10.07 8.50216 12Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`Icons|Action Sort 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <svg
      height={24}
      onClick={[Function]}
      style={
        Object {
          "color": "",
        }
      }
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        clipRule="evenodd"
        d="M14.46 8.28999L12 5.82999L9.53001 8.28999C9.14001 8.67999 8.51001 8.67999 8.12001 8.28999C7.73001 7.89999 7.73001 7.26999 8.12001 6.87999L11.29 3.69999C11.4768 3.51273 11.7305 3.4075 11.995 3.4075C12.2595 3.4075 12.5132 3.51273 12.7 3.69999L15.87 6.87999C16.26 7.26999 16.26 7.89999 15.87 8.28999C15.48 8.67999 14.85 8.67999 14.46 8.28999ZM9.54001 15.71L12 18.17L14.47 15.72C14.6568 15.5327 14.9105 15.4275 15.175 15.4275C15.4395 15.4275 15.6932 15.5327 15.88 15.72C16.27 16.11 16.27 16.74 15.88 17.13L12.71 20.3C12.32 20.69 11.69 20.69 11.3 20.3L8.13001 17.12C7.74001 16.73 7.74001 16.1 8.13001 15.71C8.31685 15.5227 8.5705 15.4175 8.83501 15.4175C9.09953 15.4175 9.35318 15.5227 9.54001 15.71Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;
